Scripts/Utilities/TransitionUtils.cs

```diff
@@ -0,0 +1,71 @@
+﻿#if UNITY_EDITOR
+using System;
+using System.Reflection;
+using UnityEditor.Experimental.EditorVR.Modules;
+using UnityEditor.Experimental.EditorVR.UI;
+using UnityEngine;
+
+namespace UnityEditor.Experimental.EditorVR.Utilities
+{
+    /// <summary>
+    /// Transition utilities
+    /// </summary>
+    public static class TransitionUtils
+    {
+        public static void AnimateProperty<T>(float time, bool state, ref bool lastState, ref float changeTime,
+            ref T property, ref T propertyStart, T startValue, T endValue, float totalDuration,
+            Func<T, T, bool> approximately, Func<T, T, T, float> getPercentage, Func<T, T, float, T> lerp,
+            Action<T> setProperty, bool setState = true, Action<T> complete = null) where T : struct
+        {
+            if (state != lastState)
+            {
+                changeTime = time;
+                propertyStart = property;
+            }
+
+            var timeDiff = time - changeTime;
+            var source = state ? startValue : endValue;
+            var target = state ? endValue : startValue;
+            if (!approximately(property, target))
+            {
+                var duration = (1 - getPercentage(propertyStart, target, source)) * totalDuration;
+                var smoothedAmount = MathUtilsExt.SmoothInOutLerpFloat(timeDiff / duration);
+                if (smoothedAmount > 1)
```

Comment:
'>=?'

Reply:
Looks like that is the approach used for the hard-set, so the value meets, or matches the target as closely as possible (depending on delta) before the hard-set.

Reply:
Yes. The == 1 is handled by the approximately check above. If we do miraculously have a value of 1, we'll get the same result from the lerp below.